pkg/backend/display/rows.go

```diff
@@ -229,6 +234,33 @@ func (data *resourceRowData) RecordDiagEvent(event engine.Event) {
 	}
 }
 
+func (data *resourceRowData) RecordPolicyViolationEvent(event engine.Event) {
+
+	//
+	// NOTE: The display code only understands DiagEvents. We convert the policy violation events
+	// accordingly so they can be displayed.
```

Comment:
Just wanted to call this out as a weird thing that I chose to do in this review because I thought it would be simpler.